crates/common/mqtt_channel/src/connection.rs

```diff
@@ -200,8 +199,7 @@ impl Connection {
                     }
                 }
 
-                Ok(Event::Incoming(Incoming::Disconnect))
-                | Ok(Event::Outgoing(Outgoing::Disconnect)) => {
+                Ok(Event::Incoming(Incoming::Disconnect)) => {
```

Comment:
Could you explain this change? Why is `Outgoing::Disconnect` ignored?

Reply:
Exiting on `Event::Outgoing(Outgoing::Disconnect)` prevents a `client.disconnect()` to be effective - since not forwarded by the `rumqttc` state machine. The impacts were mostly on the tests related to last will messages: this is the only places with explicit `client.disconnect()` calls.